parking_permits/views.py

```diff
@@ -330,12 +335,65 @@ def post(self, request, format=None):
                     permit.create_parkkihubi_permit()
 
         logger.info(f"{order} is confirmed and order permits are set to VALID ")
+        return Response({"message": "Payment received"}, status=200)
+
+
+class OrderView(APIView):
+    @swagger_auto_schema(
+        operation_description="Process order notifications.",
+        request_body=OrderSerializer,
+        security=[],
+        responses={
+            200: openapi.Response("Order received response", MessageResponseSerializer)
+        },
+        tags=["Order"],
+    )
+    @transaction.atomic
+    def post(self, request, format=None):
+        logger.info(f"Order received. Data = {json.dumps(request.data, default=str)}")
+        talpa_order_id = request.data.get("orderId")
+        talpa_subscription_id = request.data.get("subscriptionId")
+        event_type = request.data.get("eventType")
+
+        if not talpa_order_id:
+            logger.error("Talpa order id is missing from request data")
+            return Response({"message": "No order id is provided"}, status=400)
+        if not talpa_subscription_id:
+            logger.error("Talpa subscription id is missing from request data")
+            return Response({"message": "No subscription id is provided"}, status=400)
+
+        # Subscriptipn renewal process
+        if event_type == "SUBSCRIPTION_RENEWAL_ORDER_CREATED":
+            subscription = Subscription.objects.get(
+                talpa_subscription_id=talpa_subscription_id
+            )
+            permit = subscription.permit
+            permit.end_time = permit.end_time + relativedelta(months=1)
+            permit.save()
+            order = Order.objects.create(
+                talpa_order_id=talpa_order_id,
+                status=OrderStatus.CONFIRMED,
+                payment_type=OrderPaymentType.ONLINE_PAYMENT,
+                customer=permit.customer,
+            )
+            order.permits.add(permit)
+            order.subscriptions.add(subscription)
+            order.save()
+            send_permit_email(PermitEmailType.UPDATED, permit)
+            try:
+                permit.update_parkkihubi_permit()
+            except ParkkihubiPermitError:
+                permit.create_parkkihubi_permit()
+            logger.info(
+                f"{subscription} is renewed and permit end time is set to {permit.end_time}"
+            )
+            return Response({"message": "Subscription renewal completed"}, status=200)
         return Response({"message": "Order received"}, status=200)
```

Comment:
Should this default be something alse than 200?

Reply:
Changed to use error code 400 in each Talpa webhook when `event_type` is not valid.